### PR TITLE
Add google go template

### LINKED ISSUE #5724: Add Support for Go in Google Cloud Functions
# This is a Feature Proposal

## Add Support for Go in Google Cloud Functions

On 16 January 2019, [Google announced support for Go v1.11 in Cloud Functions](https://cloud.google.com/blog/products/application-development/cloud-functions-go-1-11-is-now-a-supported-language). 

Support for Go in Google Cloud Functions via `serverless create --template google-go` is proposed. 

One way to accomplish this is by integrating existing functionality from `lib/plugin/create/templates/aws-go-mod` and `lib/plugin/create/templates/google-nodejs`.

* What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
  * As a Golang developer and user of Google Cloud Functions (GCF) and the Serverless Framework, I want a template for Go on GCF in order to create functions written in Go using the Serverless Framework on GCF.
* If there is additional config how would it look
  * No additional config is proposed

Similar or dependent issues:
* &lt;None found&gt;

### PR DESCRIPTION
**Pull request by toshi0607**
## What did you implement:

close #5724

Cloud Functions Go runtime has bees released! 🎉 So I wrote a template for it.

## How did you implement it:

I added template, list of template and tests.

## How can we verify it:

```
# Go 1.11 is necessary
# $GOPATH should be set

$ cd $GOPATH/src/github.com/[your repo]
$ serverless create -t google-go -p sample-app
$ cd sample-app
$ make gomodgen
```
## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by exoego** [comment]
:+1:

This closes 
https://github.com/serverless/serverless/issues/5724

File: docker-compose.yml [hunk 1]
```diff
@@ -102,6 +102,10 @@ services:
     image: node:6.9.1
     volumes:
       - ./tmp/serverless-integration-test-google-nodejs:/app
+  google-nodejs:
```
Comment (pmuens) [inline docker-compose.yml]: Shouldn't this be `google-go` or `google-golang`?

File: lib/plugins/create/templates/google-go/fn_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"testing"
+)
+
+func TestHello(t *testing.T) {
+	tests := map[string]struct {
+		name       string
+		wantStatus int
+		wantString string
+	}{
+		"name specified":     {"toshi0607", http.StatusOK, "Hello, toshi0607!"},
```
Comment (pmuens) [inline lib/plugins/create/templates/google-go/fn_test.go]: Could you make this more general like `jdoe` or something like that?

**Comment by toshi0607** [comment]
@pmuens Thanks! I fixed issues.

File: docker-compose.yml [hunk 3]
```diff
@@ -102,6 +102,10 @@ services:
     image: node:6.9.1
     volumes:
       - ./tmp/serverless-integration-test-google-nodejs:/app
+  google-go:
+    image: golang:1.11
+    volumes:
+      - ./tmp/serverless-integration-test-google-go:/app
```
Comment (sebito91) [inline docker-compose.yml]: Should also need the `$GOPATH` location as well, look at `aws-go`:

```
  google-go:
    image: golang:1.11
    volumes:
      - ./tmp/serverless-integration-test-google-go:/app
      - ./tmp/serverless-integration-test-google:/go/src/app
```

File: lib/plugins/create/templates/google-go/serverless.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+service: gcf-go111 # NOTE: Don't put the word "google" in here
+
+provider:
+  name: google
+  runtime: go111
+  project: my-project
+  # the path to the credentials file needs to be absolute
```
Comment (sebito91) [inline lib/plugins/create/templates/google-go/serverless.yml]: Would add a reference to the docs as we have in the `examples` repo:

```

# The GCF credentials can be a little tricky to set up. Luckily we've documented this for you here:
# https://serverless.com/framework/docs/providers/google/guide/credentials/
# 
# the path to the credentials file needs to be absolute
```

Comment (toshi0607) [reply lib/plugins/create/templates/google-go/serverless.yml]: Great docs! I also added to the node one.

**Pull request merged**